packages/fuels-core/src/abi_encoder.rs

```diff
@@ -65,8 +66,19 @@ impl ABIEncoder {
         self.encode_tokens(arg_array)
     }
 
-    fn encode_string(&mut self, arg_string: &str) {
-        self.buffer.extend(pad_string(arg_string));
+    fn encode_string(&mut self, arg_string: &StringToken) {
+        if !arg_string.0.is_ascii() {
+            panic!("String parameters can only have ascii values");
+        }
+
+        if arg_string.0.len() != arg_string.1 {
```

Comment:
Readability might benefit from having the tuple restructured into descriptively named variables:
```rust
let (str_data, expected_len) = arg_string;

if str_data.len() != expected_len { ...
```

Perhaps even a reason for using a struct instead of a tuple:
```rust
struct StringToken {
    data: String,
    expected_len: usize,
}

impl StringToken {
    pub fn correct_len(&self) -> bool {
        self.data.len() == self.expected_len
    }
}

// later on
if !string_arg.correct_len() {
    // ...
}

```